ts/src/bingx.ts

```diff
@@ -0,0 +1,2342 @@
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": {
+        //            "longLeverage": 6,
+        //            "shortLeverage": 6
+        //        }
+        //    }
+        //
+        return response;
+    }
+
+    async setLeverage (leverage, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#setLeverage
+         * @description set the level of leverage for a market
+         * @see https://bingx-api.github.io/docs/swapV2/trade-api.html#_12-switch-leverage
+         * @param {float} leverage the rate of leverage
+         * @param {string} symbol unified market symbol
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} response from the exchange
+         */
+        this.checkRequiredSymbol ('setLeverage', symbol);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const side = this.safeStringUpper (params, 'side');
+        this.checkRequiredArgument ('setLeverage', side, 'side');
+        const request = {
+            'symbol': market['id'],
+            'side': side,
+            'leverage': leverage,
+        };
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": {
+        //            "leverage": 6,
+        //            "symbol": "BTC-USDT"
+        //        }
+        //    }
+        //
+        return await this.swapV2PrivatePostTradeLeverage (this.extend (request, params));
+    }
+
+    sign (path, section = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const type = section[0];
+        const version = section[1];
+        const access = section[2];
+        let url = this.implodeHostname (this.urls['api'][type]);
+        if (type === 'spot' && version === 'v3') {
+            url += '/api';
+        } else {
+            url += '/' + type;
+        }
+        url += '/' + version + '/';
+        path = this.implodeParams (path, params);
+        url += path;
+        params = this.omit (params, this.extractParams (path));
+        params = this.keysort (params);
+        if (access === 'public') {
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        } else if (access === 'private') {
+            this.checkRequiredCredentials ();
+            params['timestamp'] = this.nonce ();
+            let query = this.urlencode (params);
+            const signature = this.hmac (this.encode (query), this.encode (this.secret), sha256);
+            if (Object.keys (params).length) {
+                query = '?' + query + '&';
+            } else {
+                query += '?';
+            }
+            query += 'signature=' + signature;
+            headers = {
```

Comment:
@AlgoQ can we add this in the headers?
```
'X-SOURCE-KEY: CCXT'
```

Reply:
@carlosmiei Currently we only add headers when we use a private function, I just add it here? Or do I always add it (generally - so also when we use public endpoints)?

Reply:
@AlgoQ yeah it is ok adding it on private endpoints only